File: src/main/java/gg/mineral/server/network/packet/play/clientbound/AttachEntityPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.server.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public record AttachEntityPacket(int entityId, int vehicleId, boolean leash) implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        os.writeInt(entityId);
        os.writeInt(vehicleId);
        os.writeBoolean(leash);
    }

    @Override
    public byte getId() {
        return 0x1B;
    }
}

File: server/src/main/kotlin/gg/mineral/server/network/packet/play/clientbound/AttachEntityPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class AttachEntityPacket(val entityId: Int, val vehicleId: Int, val leash: Boolean) : Packet.Outgoing {
    override fun serialize(os: ByteBuf) {
        os.writeInt(entityId, vehicleId)
        os.writeBoolean(leash)
    }

    override val id: Byte
        get() = 0x1B
}
